netns_unspecified.go

```diff
@@ -0,0 +1,19 @@
+// +build !linux
+
+package netlink
+
+func GetNetNsIdByPid(pid int) (int, error) {
+	return -1, ErrNotImplemented
```

Comment:
IMO it's more common to return a zero value instead of -1 in case of error.
Setting there -1 would be more like double error indication with false information.

Reply:
The challenge is that 0 is a valid namespace ID.

Reply:
I think @jellonek comment is more about current diff overloads the `int` return param having it signal an error. Which is unnecessary and confusing for the reader of the code.

Given the function (correctly in go world) returns an error param, caller must honor it and not even look or use the other return params in case err != nil. The double error marking is unnecessary and confusing.